balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -232,6 +238,10 @@ func (b *pickfirstBalancer) UpdateClientConnState(state balancer.ClientConnState
 	// SubConn multiple times in the same pass. We don't want this.
 	newAddrs = deDupAddresses(newAddrs)
 
+	// Interleave addresses of both families (IPv4 and IPv6) as per RFC-8305
+	// section 4.
```

Comment:
Section 4 does talk about some stuff that we don't do like maintain RTT state etc. And I guess the "Destination Address Selection" mentioned in that section is already done by the DNS resolver implementation in the stdlib, so we don't do that either. Is that right?

In any case, it might be worth adding a function level docstring for `interleaveAddresses` which links to RFC-8305-section-4, but also documents what is *being* done in the function at a high level. If we do that, we won't need this comment here. 

Thanks.